server/src/main/java/org/apache/iotdb/db/query/dataset/AlignByDeviceDataSet.java

```diff
@@ -127,6 +130,15 @@ public AlignByDeviceDataSet(
         this.rawDataQueryPlan.setEnableRedirect(alignByDevicePlan.isEnableRedirect());
     }
 
+    int index = 1; // start 1, because first is device name
+    if (withoutNullColumnsIndex.isEmpty()) {
+      for (String measurement : this.measurements) {
```

Comment:
this measurement doesn't contain alias, but withoutNullColumns already replaced with alias, so you should use `measurementInfoMap` to get the alias.

Reply:
fix